lib/fizzy/leb128.hpp

```diff
@@ -2,8 +2,18 @@
 
 namespace fizzy
 {
-inline constexpr uint64_t leb128_decode(const uint8_t*) noexcept
+inline constexpr uint64_t leb128u_decode(const uint8_t* input) noexcept
 {
-    return 0;
+    uint64_t result = 0;
+    uint8_t result_shift = 0;
+
+    for (auto byte = input;; ++byte, result_shift += 7)
+    {
+        result |= static_cast<uint64_t>(*byte & 0x7F) << result_shift;
```

Comment:
The result value can overflow here. How to handle this?

Reply:
It looks like according to the spec it would be also malformed encoding 
https://webassembly.github.io/spec/core/binary/values.html#bytes

>  As an additional constraint, the total number of bytes encoding a value of type uN must not exceed ceil(N/7) bytes.